test/modules/TestResultCoordinator/Reports/DirectMethod/DirectMethodTestResultComparer.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator.DirectMethod
+{
+    using System;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Newtonsoft.Json;
+    using TestResultCoordinator.Reports;
+
+    /// <summary>
+    /// This test result comparer is used to determine if 2 TestOperationResult instances matches.
+    /// </summary>
+    public sealed class DirectMethodTestOperationResultComparer : ITestResultComparer<TestOperationResult>
+    {
+        public bool Matches(TestOperationResult value1, TestOperationResult value2)
+        {
+            if ((value1 == null && value2 != null) || (value1 != null && value2 == null))
+            {
+                return false;
+            }
+
+            if (value1 == null && value2 == null)
+            {
+                return true;
+            }
+
+            DirectMethodTestResult dmResult1 = JsonConvert.DeserializeObject<DirectMethodTestResult>(value1.Result);
+            DirectMethodTestResult dmResult2 = JsonConvert.DeserializeObject<DirectMethodTestResult>(value2.Result);
+
+            return dmResult1.TrackingId == dmResult2.TrackingId
+                && dmResult1.SequenceNumber == dmResult2.SequenceNumber
+                && dmResult1.HttpStatusCode == dmResult2.HttpStatusCode;
+        }
+    }
+}
```

Comment:
Made this comparer so that we can use in the filter abstracton. Before this PR, it wasn't needed because we make assumptions about how dm tests operate (i.e. receiver never sends any failure result).

I made this comparer and only integrated into the filter, not the report generator which works as before.